src/System.Net.Security/src/System/Net/SecureProtocols/_SecuritySafeHandles.Windows.cs

```diff
@@ -0,0 +1,1340 @@
+                                                    ref authdata,
+                                                    null,
+                                                    null,
+                                                    ref outCredential._handle,
+                                                    out timeStamp
+                                                    );
+            }
+            finally
+            {
+                authdata.certContextArray = copiedPtr;
+            }
+
+#if TRAVE
+            GlobalLog.Print("Unmanaged::AcquireCredentialsHandle() returns 0x"
+                            + errorCode.ToString("x")
+                            + ", handle = " + outCredential.ToString()
+                            );
+#endif
+
+            if (errorCode != 0)
+            {
+                outCredential.SetHandleAsInvalid();
+            }
+            return errorCode;
+        }
+    }
+
+    //
+    // This is a class holding a Credential handle reference, used for static handles cache
+    //
+#if DEBUG
+    internal sealed class SafeCredentialReference : DebugCriticalHandleMinusOneIsInvalid
+    {
+#else
+    internal sealed class SafeCredentialReference : CriticalHandleMinusOneIsInvalid
+    {
+#endif
+
+        //
+        // Static cache will return the target handle if found the reference in the table.
+        //
+        internal SafeFreeCredentials _Target;
+
+        internal static SafeCredentialReference CreateReference(SafeFreeCredentials target)
+        {
+            SafeCredentialReference result = new SafeCredentialReference(target);
+            if (result.IsInvalid)
+                return null;
+
+            return result;
+        }
+        private SafeCredentialReference(SafeFreeCredentials target) : base()
+        {
+            // Bumps up the refcount on Target to signify that target handle is statically cached so
+            // its dispose should be postponed
+            bool b = false;
+            try
+            {
+                target.DangerousAddRef(ref b);
+            }
+            catch
+            {
+                if (b)
+                {
+                    target.DangerousRelease();
+                    b = false;
+                }
+            }
+            finally
+            {
+                if (b)
+                {
+                    _Target = target;
+                    SetHandle(new IntPtr(0));   // make this handle valid
+                }
+            }
+        }
+
+        override protected bool ReleaseHandle()
+        {
+            SafeFreeCredentials target = _Target;
+            if (target != null)
+                target.DangerousRelease();
+            _Target = null;
+            return true;
+        }
+    }
+
+    internal sealed class SafeFreeCredential_SECURITY : SafeFreeCredentials
+    {
+        public SafeFreeCredential_SECURITY() : base() { }
+
+        override protected bool ReleaseHandle()
```

Comment:
Yes. Again, the automatic tooling didn't detect/change this. I'll search for other places.